Slider/Assets/Scripts/Sliders/Grid/AreaGrids/MagiTechGrid.cs

```diff
@@ -81,39 +90,129 @@ public static bool IsInPast(Transform transform)
 
     #endregion
 
+    private List<Collider2D> GetCollidingItems(GameObject obj)
+    {
+        ContactFilter2D filter = new()
+        {
+            layerMask = LayerMask.GetMask("Item"),
+            useLayerMask = true
+        };
+
+        BoxCollider2D collider = obj.GetComponent<BoxCollider2D>();
+        List<Collider2D> list = new();
+        collider.OverlapCollider(filter, list);
+        return list;
+    }
+
     public void HasTwoBurgers(Condition c)
     {
-        if (desyncBurger.IsDesynced)
+        if (!desyncBurger.IsDesynced)
         {
-            BoxCollider2D collider = hungryBoi.GetComponent<BoxCollider2D>();
+            c.SetSpec(false);
+            return;
+        }
 
-            ContactFilter2D filter = new ContactFilter2D();
-            filter.layerMask = LayerMask.GetMask("Item");
-            filter.useLayerMask = true;
+        bool hasBurger = false;
+        bool hasDesyncBurger = false;
 
-            List<Collider2D> list = new List<Collider2D>(); //Change this to however many items can be put inside hungry boi's collider
-            collider.OverlapCollider(filter, list);
+        foreach (Collider2D hit in GetCollidingItems(hungryBoi.gameObject))
```

Comment:
if you want you can prob handle the code checking for if gameobjects are in the colliders in `Update()` and set a bool, you can look at how `CatTrick.cs` does it in village (might be a bit cleaner)